src/Shared/CertificateGeneration/UnixCertificateManager.cs

```diff
@@ -18,13 +39,75 @@ internal UnixCertificateManager(string subject, int version)
 
-    public override bool IsTrusted(X509Certificate2 certificate)
+    public override TrustLevel GetTrustLevel(X509Certificate2 certificate)
     {
-        using X509Chain chain = new X509Chain();
+        var sawTrustSuccess = false;
+        var sawTrustFailure = false;
+
+        if (!string.IsNullOrEmpty(Environment.GetEnvironmentVariable(OpenSslCertDirectoryOverrideVariableName)))
+        {
+            // Warn but don't bail.
+            Log.UnixOpenSslCertificateDirectoryOverrideIgnored(OpenSslCertDirectoryOverrideVariableName);
+        }
+
+        // Building the chain will check whether openssl (which covers HttpClient) trusts the cert.
+        // An alternative approach would be to look for the file and link in the trust folder, but
+        // this tests the real-world behavior.
+        using var chain = new X509Chain();
         // This is just a heuristic for whether or not we should prompt the user to re-run with `--trust`
         // so we don't need to check revocation (which doesn't really make sense for dev certs anyway)
         chain.ChainPolicy.RevocationMode = X509RevocationMode.NoCheck;
-        return chain.Build(certificate);
+        if (chain.Build(certificate))
+        {
+            sawTrustSuccess = true;
+        }
+        else
+        {
+            sawTrustFailure = true;
+
+            if (string.IsNullOrEmpty(Environment.GetEnvironmentVariable(OpenSslCertificateDirectoryVariableName)))
+            {
+                Log.UnixNotTrustedByOpenSslVariableUnset(OpenSslCertificateDirectoryVariableName);
+            }
+            else
+            {
+                Log.UnixNotTrustedByOpenSsl();
+            }
+        }
+
+        var nssDbs = GetNssDbs(Environment.GetFolderPath(Environment.SpecialFolder.UserProfile));
+        if (nssDbs.Count > 0)
+        {
+            if (!IsCommandAvailable(CertUtilCommand))
+            {
+                // If there are browsers but we don't have certutil, we can't check trust and,
+                // in all probability, we can't have previously established it.
+                Log.UnixMissingCertUtilCommand(CertUtilCommand);
+                sawTrustFailure = true;
+            }
+            else
+            {
+                var nickname = GetCertificateNickname(certificate);
+                foreach (var nssDb in nssDbs)
+                {
+                    if (IsCertificateInNssDb(nickname, nssDb))
+                    {
+                        sawTrustSuccess = true;
+                    }
+                    else
+                    {
+                        sawTrustFailure = true;
+                        Log.UnixNotTrustedByNss(nssDb.Path, nssDb.IsFirefox ? BrowserFamilyFirefox : BrowserFamilyChromium);
+                    }
+                }
+            }
+        }
+
+        return sawTrustSuccess
+            ? sawTrustFailure
+                ? TrustLevel.Partial
+                : TrustLevel.Full
+            : TrustLevel.None;
```

Comment:
This tripped me up a bit when I read it, might be worth putting it in terms of the two things we are checking (openssl and nssdb) and maybe breaking it down into a couple of `if`s with a return statement.